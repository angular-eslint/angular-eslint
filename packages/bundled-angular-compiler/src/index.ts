// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck

export * from '@angular/compiler';

import type {
  MessageBundle,
  ParsedTemplate,
  LocalizedString,
} from '@angular/compiler';
export type Comment = Required<ParsedTemplate>['commentNodes'][number];
export type Message = ReturnType<MessageBundle['getMessages']>[number];
export type I18nMeta = LocalizedString['metaBlock'];
