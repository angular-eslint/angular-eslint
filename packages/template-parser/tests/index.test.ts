/**
 * This usage seems a little wonky, but don't want to run any risk of breaking changes
 * to the compiled output of the template-parser node module, so leaving until v13.
 */
import { default as templateParser } from '../src/index';

describe('parseForESLint()', () => {
  it('should work', () => {
    expect(
      templateParser.parseForESLint(
        `
      <!-- eslint-disable-next-line -->
      <div>some node</div>
      <!-- some other comment -->
    `,
        { filePath: './foo.html' },
      ).ast,
    ).toMatchInlineSnapshot(`
      Object {
        "comments": Array [
          Object {
            "loc": Object {
              "end": Object {
                "column": 10,
                "line": 2,
              },
              "start": Object {
                "column": 6,
                "line": 2,
              },
            },
            "range": Array [
              7,
              11,
            ],
            "type": "Block",
            "value": "eslint-disable-next-line",
          },
          Object {
            "loc": Object {
              "end": Object {
                "column": 10,
                "line": 4,
              },
              "start": Object {
                "column": 6,
                "line": 4,
              },
            },
            "range": Array [
              74,
              78,
            ],
            "type": "Block",
            "value": "some other comment",
          },
        ],
        "loc": Object {
          "end": Object {
            "column": 4,
            "line": 5,
          },
          "start": Object {
            "column": 6,
            "line": 2,
          },
        },
        "range": Array [
          7,
          106,
        ],
        "templateNodes": Array [
          Text {
            "sourceSpan": ParseSourceSpan {
              "details": null,
              "end": ParseLocation {
                "col": 6,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 1,
                "offset": 7,
              },
              "fullStart": ParseLocation {
                "col": 0,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 0,
                "offset": 0,
              },
              "start": ParseLocation {
                "col": 6,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 1,
                "offset": 7,
              },
            },
            "type": "Text",
            "value": "
            ",
          },
          Text {
            "sourceSpan": ParseSourceSpan {
              "details": null,
              "end": ParseLocation {
                "col": 6,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 2,
                "offset": 47,
              },
              "fullStart": ParseLocation {
                "col": 39,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 1,
                "offset": 40,
              },
              "start": ParseLocation {
                "col": 6,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 2,
                "offset": 47,
              },
            },
            "type": "Text",
            "value": "
            ",
          },
          Element {
            "attributes": Array [],
            "children": Array [
              Text {
                "sourceSpan": ParseSourceSpan {
                  "details": null,
                  "end": ParseLocation {
                    "col": 20,
                    "file": ParseSourceFile {
                      "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                      "url": "./foo.html",
                    },
                    "line": 2,
                    "offset": 61,
                  },
                  "fullStart": ParseLocation {
                    "col": 11,
                    "file": ParseSourceFile {
                      "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                      "url": "./foo.html",
                    },
                    "line": 2,
                    "offset": 52,
                  },
                  "start": ParseLocation {
                    "col": 11,
                    "file": ParseSourceFile {
                      "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                      "url": "./foo.html",
                    },
                    "line": 2,
                    "offset": 52,
                  },
                },
                "type": "Text",
                "value": "some node",
              },
            ],
            "endSourceSpan": ParseSourceSpan {
              "details": null,
              "end": ParseLocation {
                "col": 26,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 2,
                "offset": 67,
              },
              "fullStart": ParseLocation {
                "col": 20,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 2,
                "offset": 61,
              },
              "start": ParseLocation {
                "col": 20,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 2,
                "offset": 61,
              },
            },
            "i18n": undefined,
            "inputs": Array [],
            "name": "div",
            "outputs": Array [],
            "references": Array [],
            "sourceSpan": ParseSourceSpan {
              "details": null,
              "end": ParseLocation {
                "col": 26,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 2,
                "offset": 67,
              },
              "fullStart": ParseLocation {
                "col": 6,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 2,
                "offset": 47,
              },
              "start": ParseLocation {
                "col": 6,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 2,
                "offset": 47,
              },
            },
            "startSourceSpan": ParseSourceSpan {
              "details": null,
              "end": ParseLocation {
                "col": 11,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 2,
                "offset": 52,
              },
              "fullStart": ParseLocation {
                "col": 6,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 2,
                "offset": 47,
              },
              "start": ParseLocation {
                "col": 6,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 2,
                "offset": 47,
              },
            },
            "type": "Element",
          },
          Text {
            "sourceSpan": ParseSourceSpan {
              "details": null,
              "end": ParseLocation {
                "col": 6,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 3,
                "offset": 74,
              },
              "fullStart": ParseLocation {
                "col": 26,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 2,
                "offset": 67,
              },
              "start": ParseLocation {
                "col": 6,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 3,
                "offset": 74,
              },
            },
            "type": "Text",
            "value": "
            ",
          },
          Text {
            "sourceSpan": ParseSourceSpan {
              "details": null,
              "end": ParseLocation {
                "col": 4,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 4,
                "offset": 106,
              },
              "fullStart": ParseLocation {
                "col": 33,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 3,
                "offset": 101,
              },
              "start": ParseLocation {
                "col": 4,
                "file": ParseSourceFile {
                  "content": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
                  "url": "./foo.html",
                },
                "line": 4,
                "offset": 106,
              },
            },
            "type": "Text",
            "value": "
          ",
          },
        ],
        "tokens": Array [],
        "type": "Program",
        "value": "
            <!-- eslint-disable-next-line -->
            <div>some node</div>
            <!-- some other comment -->
          ",
      }
    `);
  });
});
