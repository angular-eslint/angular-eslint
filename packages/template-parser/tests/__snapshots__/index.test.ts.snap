// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseForESLint() ng-content should support fallback content 1`] = `
Object {
  "comments": Array [],
  "loc": Object {
    "end": Object {
      "column": 8,
      "line": 7,
    },
    "start": Object {
      "column": 10,
      "line": 2,
    },
  },
  "range": Array [
    11,
    129,
  ],
  "templateNodes": Array [
    Text$3 {
      "loc": Object {
        "end": Object {
          "column": 10,
          "line": 2,
        },
        "start": Object {
          "column": 10,
          "line": 2,
        },
      },
      "sourceSpan": ParseSourceSpan {
        "details": null,
        "end": ParseLocation {
          "col": 10,
          "file": ParseSourceFile {
            "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
            "url": "./foo.html",
          },
          "line": 1,
          "offset": 11,
        },
        "fullStart": ParseLocation {
          "col": 0,
          "file": ParseSourceFile {
            "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
            "url": "./foo.html",
          },
          "line": 0,
          "offset": 0,
        },
        "start": ParseLocation {
          "col": 10,
          "file": ParseSourceFile {
            "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
            "url": "./foo.html",
          },
          "line": 1,
          "offset": 11,
        },
      },
      "type": "Text$3",
      "value": "
          ",
    },
    Content {
      "attributes": Array [],
      "children": Array [
        Text$3 {
          "loc": Object {
            "end": Object {
              "column": 13,
              "line": 3,
            },
            "start": Object {
              "column": 13,
              "line": 3,
            },
          },
          "sourceSpan": ParseSourceSpan {
            "details": null,
            "end": ParseLocation {
              "col": 13,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 2,
              "offset": 37,
            },
            "fullStart": ParseLocation {
              "col": 22,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 1,
              "offset": 23,
            },
            "start": ParseLocation {
              "col": 13,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 2,
              "offset": 37,
            },
          },
          "type": "Text$3",
          "value": "
             ",
        },
        IfBlock {
          "branches": Array [
            IfBlockBranch {
              "children": Array [
                Text$3 {
                  "loc": Object {
                    "end": Object {
                      "column": 15,
                      "line": 4,
                    },
                    "start": Object {
                      "column": 15,
                      "line": 4,
                    },
                  },
                  "sourceSpan": ParseSourceSpan {
                    "details": null,
                    "end": ParseLocation {
                      "col": 15,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 3,
                      "offset": 64,
                    },
                    "fullStart": ParseLocation {
                      "col": 24,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 2,
                      "offset": 48,
                    },
                    "start": ParseLocation {
                      "col": 15,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 3,
                      "offset": 64,
                    },
                  },
                  "type": "Text$3",
                  "value": "
               ",
                },
                Element$1 {
                  "attributes": Array [],
                  "children": Array [
                    Text$3 {
                      "loc": Object {
                        "end": Object {
                          "column": 26,
                          "line": 4,
                        },
                        "start": Object {
                          "column": 20,
                          "line": 4,
                        },
                      },
                      "sourceSpan": ParseSourceSpan {
                        "details": null,
                        "end": ParseLocation {
                          "col": 26,
                          "file": ParseSourceFile {
                            "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                            "url": "./foo.html",
                          },
                          "line": 3,
                          "offset": 75,
                        },
                        "fullStart": ParseLocation {
                          "col": 20,
                          "file": ParseSourceFile {
                            "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                            "url": "./foo.html",
                          },
                          "line": 3,
                          "offset": 69,
                        },
                        "start": ParseLocation {
                          "col": 20,
                          "file": ParseSourceFile {
                            "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                            "url": "./foo.html",
                          },
                          "line": 3,
                          "offset": 69,
                        },
                      },
                      "type": "Text$3",
                      "value": "Hello!",
                    },
                  ],
                  "endSourceSpan": ParseSourceSpan {
                    "details": null,
                    "end": ParseLocation {
                      "col": 32,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 3,
                      "offset": 81,
                    },
                    "fullStart": ParseLocation {
                      "col": 26,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 3,
                      "offset": 75,
                    },
                    "start": ParseLocation {
                      "col": 26,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 3,
                      "offset": 75,
                    },
                  },
                  "i18n": undefined,
                  "inputs": Array [],
                  "loc": Object {
                    "end": Object {
                      "column": 32,
                      "line": 4,
                    },
                    "start": Object {
                      "column": 15,
                      "line": 4,
                    },
                  },
                  "name": "div",
                  "outputs": Array [],
                  "references": Array [],
                  "sourceSpan": ParseSourceSpan {
                    "details": null,
                    "end": ParseLocation {
                      "col": 32,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 3,
                      "offset": 81,
                    },
                    "fullStart": ParseLocation {
                      "col": 15,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 3,
                      "offset": 64,
                    },
                    "start": ParseLocation {
                      "col": 15,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 3,
                      "offset": 64,
                    },
                  },
                  "startSourceSpan": ParseSourceSpan {
                    "details": null,
                    "end": ParseLocation {
                      "col": 20,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 3,
                      "offset": 69,
                    },
                    "fullStart": ParseLocation {
                      "col": 15,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 3,
                      "offset": 64,
                    },
                    "start": ParseLocation {
                      "col": 15,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 3,
                      "offset": 64,
                    },
                  },
                  "type": "Element$1",
                },
                Text$3 {
                  "loc": Object {
                    "end": Object {
                      "column": 13,
                      "line": 5,
                    },
                    "start": Object {
                      "column": 13,
                      "line": 5,
                    },
                  },
                  "sourceSpan": ParseSourceSpan {
                    "details": null,
                    "end": ParseLocation {
                      "col": 13,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 4,
                      "offset": 95,
                    },
                    "fullStart": ParseLocation {
                      "col": 32,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 3,
                      "offset": 81,
                    },
                    "start": ParseLocation {
                      "col": 13,
                      "file": ParseSourceFile {
                        "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                        "url": "./foo.html",
                      },
                      "line": 4,
                      "offset": 95,
                    },
                  },
                  "type": "Text$3",
                  "value": "
             ",
                },
              ],
              "endSourceSpan": ParseSourceSpan {
                "details": null,
                "end": ParseLocation {
                  "col": 14,
                  "file": ParseSourceFile {
                    "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                    "url": "./foo.html",
                  },
                  "line": 4,
                  "offset": 96,
                },
                "fullStart": ParseLocation {
                  "col": 13,
                  "file": ParseSourceFile {
                    "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                    "url": "./foo.html",
                  },
                  "line": 4,
                  "offset": 95,
                },
                "start": ParseLocation {
                  "col": 13,
                  "file": ParseSourceFile {
                    "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                    "url": "./foo.html",
                  },
                  "line": 4,
                  "offset": 95,
                },
              },
              "expression": ASTWithSource {
                "ast": PropertyRead {
                  "loc": Object {
                    "end": Object {
                      "column": undefined,
                      "line": NaN,
                    },
                    "start": Object {
                      "column": undefined,
                      "line": NaN,
                    },
                  },
                  "name": "foo",
                  "nameSpan": AbsoluteSourceSpan {
                    "end": 45,
                    "start": 42,
                  },
                  "receiver": ImplicitReceiver {
                    "loc": Object {
                      "end": Object {
                        "column": undefined,
                        "line": NaN,
                      },
                      "start": Object {
                        "column": undefined,
                        "line": NaN,
                      },
                    },
                    "sourceSpan": AbsoluteSourceSpan {
                      "end": 42,
                      "start": 42,
                    },
                    "span": ParseSpan {
                      "end": 0,
                      "start": 0,
                    },
                    "type": "ImplicitReceiver",
                  },
                  "sourceSpan": AbsoluteSourceSpan {
                    "end": 45,
                    "start": 42,
                  },
                  "span": ParseSpan {
                    "end": 3,
                    "start": 0,
                  },
                  "type": "PropertyRead",
                },
                "errors": Array [],
                "loc": Object {
                  "end": Object {
                    "column": undefined,
                    "line": NaN,
                  },
                  "start": Object {
                    "column": undefined,
                    "line": NaN,
                  },
                },
                "location": "./foo.html@2:18",
                "source": "foo",
                "sourceSpan": AbsoluteSourceSpan {
                  "end": 45,
                  "start": 42,
                },
                "span": ParseSpan {
                  "end": 3,
                  "start": 0,
                },
                "type": "ASTWithSource",
              },
              "expressionAlias": null,
              "i18n": undefined,
              "loc": Object {
                "end": Object {
                  "column": 14,
                  "line": 5,
                },
                "start": Object {
                  "column": 13,
                  "line": 3,
                },
              },
              "nameSpan": ParseSourceSpan {
                "details": null,
                "end": ParseLocation {
                  "col": 17,
                  "file": ParseSourceFile {
                    "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                    "url": "./foo.html",
                  },
                  "line": 2,
                  "offset": 41,
                },
                "fullStart": ParseLocation {
                  "col": 13,
                  "file": ParseSourceFile {
                    "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                    "url": "./foo.html",
                  },
                  "line": 2,
                  "offset": 37,
                },
                "start": ParseLocation {
                  "col": 13,
                  "file": ParseSourceFile {
                    "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                    "url": "./foo.html",
                  },
                  "line": 2,
                  "offset": 37,
                },
              },
              "sourceSpan": ParseSourceSpan {
                "details": null,
                "end": ParseLocation {
                  "col": 14,
                  "file": ParseSourceFile {
                    "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                    "url": "./foo.html",
                  },
                  "line": 4,
                  "offset": 96,
                },
                "fullStart": ParseLocation {
                  "col": 13,
                  "file": ParseSourceFile {
                    "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                    "url": "./foo.html",
                  },
                  "line": 2,
                  "offset": 37,
                },
                "start": ParseLocation {
                  "col": 13,
                  "file": ParseSourceFile {
                    "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                    "url": "./foo.html",
                  },
                  "line": 2,
                  "offset": 37,
                },
              },
              "startSourceSpan": ParseSourceSpan {
                "details": null,
                "end": ParseLocation {
                  "col": 24,
                  "file": ParseSourceFile {
                    "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                    "url": "./foo.html",
                  },
                  "line": 2,
                  "offset": 48,
                },
                "fullStart": ParseLocation {
                  "col": 13,
                  "file": ParseSourceFile {
                    "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                    "url": "./foo.html",
                  },
                  "line": 2,
                  "offset": 37,
                },
                "start": ParseLocation {
                  "col": 13,
                  "file": ParseSourceFile {
                    "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                    "url": "./foo.html",
                  },
                  "line": 2,
                  "offset": 37,
                },
              },
              "type": "IfBlockBranch",
            },
          ],
          "endSourceSpan": ParseSourceSpan {
            "details": null,
            "end": ParseLocation {
              "col": 14,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 4,
              "offset": 96,
            },
            "fullStart": ParseLocation {
              "col": 13,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 4,
              "offset": 95,
            },
            "start": ParseLocation {
              "col": 13,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 4,
              "offset": 95,
            },
          },
          "loc": Object {
            "end": Object {
              "column": 14,
              "line": 5,
            },
            "start": Object {
              "column": 13,
              "line": 3,
            },
          },
          "nameSpan": ParseSourceSpan {
            "details": null,
            "end": ParseLocation {
              "col": 17,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 2,
              "offset": 41,
            },
            "fullStart": ParseLocation {
              "col": 13,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 2,
              "offset": 37,
            },
            "start": ParseLocation {
              "col": 13,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 2,
              "offset": 37,
            },
          },
          "sourceSpan": ParseSourceSpan {
            "details": null,
            "end": ParseLocation {
              "col": 14,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 4,
              "offset": 96,
            },
            "fullStart": ParseLocation {
              "col": 13,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 2,
              "offset": 37,
            },
            "start": ParseLocation {
              "col": 13,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 2,
              "offset": 37,
            },
          },
          "startSourceSpan": ParseSourceSpan {
            "details": null,
            "end": ParseLocation {
              "col": 24,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 2,
              "offset": 48,
            },
            "fullStart": ParseLocation {
              "col": 13,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 2,
              "offset": 37,
            },
            "start": ParseLocation {
              "col": 13,
              "file": ParseSourceFile {
                "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
                "url": "./foo.html",
              },
              "line": 2,
              "offset": 37,
            },
          },
          "type": "IfBlock",
        },
      ],
      "i18n": undefined,
      "loc": Object {
        "end": Object {
          "column": 23,
          "line": 6,
        },
        "start": Object {
          "column": 10,
          "line": 2,
        },
      },
      "name": "ng-content",
      "selector": "*",
      "sourceSpan": ParseSourceSpan {
        "details": null,
        "end": ParseLocation {
          "col": 23,
          "file": ParseSourceFile {
            "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
            "url": "./foo.html",
          },
          "line": 5,
          "offset": 120,
        },
        "fullStart": ParseLocation {
          "col": 10,
          "file": ParseSourceFile {
            "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
            "url": "./foo.html",
          },
          "line": 1,
          "offset": 11,
        },
        "start": ParseLocation {
          "col": 10,
          "file": ParseSourceFile {
            "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
            "url": "./foo.html",
          },
          "line": 1,
          "offset": 11,
        },
      },
      "type": "Content",
    },
    Text$3 {
      "loc": Object {
        "end": Object {
          "column": 8,
          "line": 7,
        },
        "start": Object {
          "column": 8,
          "line": 7,
        },
      },
      "sourceSpan": ParseSourceSpan {
        "details": null,
        "end": ParseLocation {
          "col": 8,
          "file": ParseSourceFile {
            "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
            "url": "./foo.html",
          },
          "line": 6,
          "offset": 129,
        },
        "fullStart": ParseLocation {
          "col": 23,
          "file": ParseSourceFile {
            "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
            "url": "./foo.html",
          },
          "line": 5,
          "offset": 120,
        },
        "start": ParseLocation {
          "col": 8,
          "file": ParseSourceFile {
            "content": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
            "url": "./foo.html",
          },
          "line": 6,
          "offset": 129,
        },
      },
      "type": "Text$3",
      "value": "
        ",
    },
  ],
  "tokens": Array [],
  "type": "Program",
  "value": "
          <ng-content>
             @if (foo) {
               <div>Hello!</div>
             }
          </ng-content>
        ",
}
`;
